processor/processor.go

```diff
@@ -1561,25 +1584,54 @@ func (proc *Handle) processJobsForDest(partition string, subJobs subJob) *transf
 		statusList = append(statusList, &newStatus)
 	}
 
-	if len(eventSchemaJobs) > 0 {
-		err := misc.RetryWithNotify(
-			context.Background(),
-			proc.jobsDBCommandTimeout,
-			proc.jobdDBMaxRetries,
-			func(ctx context.Context) error {
-				return proc.eventSchemaDB.WithStoreSafeTx(
-					ctx,
-					func(tx jobsdb.StoreSafeTx) error {
-						return proc.eventSchemaDB.StoreInTx(ctx, tx, eventSchemaJobs)
-					},
-				)
-			}, proc.sendRetryStoreStats)
-		if err != nil {
-			proc.logger.Errorf("Store into event schema table failed with error: %v", err)
-			proc.logger.Errorf("eventSchemaJobs: %v", eventSchemaJobs)
-			panic(err)
+	g, ctx := errgroup.WithContext(context.Background())
+
+	g.Go(func() error {
+		if len(eventSchemaJobs) > 0 {
```

Comment:
Instead of nesting the `if` we could simplify here.

```suggestion
		if len(eventSchemaJobs) == 0 {
				return nil
		}
```